plugins/woocommerce/src/Internal/Admin/ShippingLabelBanner.php

```diff
@@ -36,40 +39,26 @@ public function __construct() {
 	 */
 	private function should_show_meta_box() {
 		if ( ! $this->shipping_label_banner_display_rules ) {
-			$jetpack_version   = null;
-			$jetpack_connected = null;
-			$wcs_version       = null;
-			$wcs_tos_accepted  = null;
-
-			if ( defined( 'JETPACK__VERSION' ) ) {
-				$jetpack_version = JETPACK__VERSION;
-			}
+			$dotcom_connected = null;
+			$wcs_version      = null;
 
 			if ( class_exists( Jetpack_Connection_Manager::class ) ) {
-				$jetpack_connected = ( new Jetpack_Connection_Manager() )->has_connected_owner();
-			}
-
-			if ( class_exists( '\WC_Connect_Loader' ) ) {
-				$wcs_version = \WC_Connect_Loader::get_wcs_version();
+				$dotcom_connected = ( new Jetpack_Connection_Manager() )->has_connected_owner();
 			}
 
-			if ( class_exists( '\WC_Connect_Options' ) ) {
-				$wcs_tos_accepted = \WC_Connect_Options::get_option( 'tos_accepted' );
+			if ( class_exists( '\Automattic\WCShipping\Utils' ) ) {
```

Comment:
I tested a "fourth" case:

[WooCommerce Shipping 1.0.5  (or lower)](https://github.com/woocommerce/woocommerce-shipping/releases/tag/1.0.5) installed, but not activated.

1. Visit an order page with one or more shippable product(s)
2. Verify you see the promotion banner
3. Click on `Create Shipping Label`
4. Verify you see the below banner:

![Screenshot 2024-08-30 at 16 50 10](https://github.com/user-attachments/assets/19161a45-8df6-4a6d-a0ed-93954b1e97bb)

**What happens:**

* **UI**: The banner will stay in its "loading" state
* **Behind the scenes**: WC Shipping was activated and it's visible if I refresh the page

I'm not sure if we want to add more resources into providing a better experience but:

1. We could add a text to inform the merchant that it might work if they refresh the page
2. We could add a check that will only show the banner if the local version of WC Shipping is 1.1.0 or newer?

Reply:
Yes, that's why we need the new change in WC Shipping, besides the banner already says the merchant needs to refresh, na?
> We could add a check that will only show the banner if the local version of WC Shipping is 1.1.0 or newer?

Well, this makes sense!

Reply:
What if in this case, we refrain from accepting TOS via ajax, and simply refresh the page? 
1) The refresh won't happen for users, let's say, in 1-2 months from now.
2) It will only happen for a fraction of users who have an older version of WC Shipping installed but not active, which should be a very small number given how new the plugin is.

What do you think?

Reply:
> We could add a check that will only show the banner if the local version of WC Shipping is 1.1.0 or newer?

> What if in this case, we refrain from accepting TOS via ajax, and simply refresh the page?

Good idea! I like both of these! Not accepting ToS definitely solves the missing endpoint not yet introduced and would help us to still expose the plugin to those with older versions.

At the same time, then I really don't think it's a lot of sites that will be affected, so IMO whatever is fastest of the two should be what we do 😄

Reply:
> besides the banner already says the merchant needs to refresh, na?

Oi! Forgot to answer this: Sorry, yes, we do say "refresh" but it also sounds like the plugin wasn't setup, but it was actually activated behind the scenes. So the most correct message for this very specific case would be some variation of: "The plugin was activated, but you need to fresh the page for it to be visible"

Reply:
Refreshing the page after activation of an incompatible version of WC Shipping is added here 84cbc6b45cbf941e6ac3928814061a47347b0da9